README.md

```diff
@@ -2,62 +2,132 @@
+
+In `[on-the-fly]` mode PRESM does not write to disk, it simply applies all the loaders specified in your `loaderconfig.json` and executes the resulting code.  
+
+In this mode, PRESM exists as a loader that you use `Node` to call:
+
+```js
+node --loader=presm foo.ts
+```
+
+### `[build]` mode
+
+In `[build]` mode PRESM first applies to loaders specified in your `loaderconfig.json`, then writes the output files to the folder specified in `outputDir` in the `loaderconfig.json`. After this mode executes, you should be able to execute these files using just `Node`.
+
+In this mode, PRESM exists as a CLI tool, `presmbuilder`; it has the following options:
+```shell
+Options:
+  --help        Show help                         [boolean]
+  --version     Show version number               [boolean]
+  -f, --file    File to build                      [string]
+  -o, --output  Directory to output; 
+                overwrites loaderconfig            [string]
+```
+#### `--file`
+This is used to target one specific file to build, instead of an entire directory.  *NOTE*:  when building one file with this option, PRESM will only apply specified loaders to this file, not its dependencie tree - therefore, all imports of that file must be bare specifiers.
+
+*Will not* work:
+```js
+single_file.ts:
+
+import myModule from './mymodule.mjs';
+import myTSModule from './mymodule.ts';
+
+myModule.doSomething();
+```
+*Will* work:
+```js
+single_file.ts:
+
+import ts from 'typescript';
+ts.doSomething();
+```
+
+#### `--output`
+This is used to override the output folder (`outputDir` in `loaderconfig.json`) that will contain the built files.  *NOTE*: this folder name cannot begin with `.` nor `/`
+
+### `loaderconfig.json`
+
+This file should exist in your projects root directory.  Below is a sample of the files format:
+```json
+{
+    "inputDir": "src",
+    "outputDir": "dist",
+    "resourceProviders": [
+      {
+        "type": "../examples/loaders/resourceprovider-basic-fs.js"
+      }
+    ],
+    "preProcessors": [
+      {
+        "name": "../examples/loaders/preprocessor-yaml.js",
+        "options": {}
+      },
+      {
+        "name": "../examples/loaders/preprocessor-typescript.js",
+        "options": {
+          "compilerOptions": {
+            "target": "esnext",
+            "module": "esnext"
+          }
+        }
+      }
+    ],
+    "postProcessors": []
+  }
+```
+- `resourceProviders`, `preProcessors`, and `postProcessors` are all list of loader objects
+- `resourceProvider` objects need a `type`
+- `pre-` and `post-` `processors` object need a `name` and can have `options` passed into the those loaders
+- All `type` and `name` files are file locations relative to the `presm` project root directory
+- If no `loaderconfig.json` exists in your proejcts root directory, `presm` will default to the `loaderconfig.json` in its `presm`'s directory
+
+## Intstallation Guide
+
+As of yet, `presm` must be cloned locally for use:
```

Comment:
I think this could be simplified to `npm install --save-dev googleinterns/presm`. `npm` allows installing packages directly from the master branch.